### DISCUSSION TITLE
ETC Support

### DISCUSSION THREAD

**Discussion by AlphaX-Qortal** [coinfoundry/miningcore#850] [Ideas]
Hello !

First of all it is not an issue more likely a request.
You deleted Ethereum but what about ETC ?
You will support it ?
What about Vertcoin which changes to Verthash ?

With best Regards
CCC

**Comment by oliverw** [comment]
The Ethereum related code was removed because it hadn't been maintained for a long time. In hindsight it will propably return inside in archive folder here in the repo. In case someone wants to pick it up.

If Verthash is going to be supported depends on the magnitude of changes required for it. But probably yes.

**Reply by github-vp** [reply]
I was able to built and run ETH on release 50 on Unix. Is anyone else able to run and mine successfully? Trying to confirm if release 50 run ETH as I have not generated any coin.

**Comment by Konstantin35** [comment]
Today I started the return of Ethereum in my [repository](https://github.com/Konstantin35/miningcore), I took Release 53 as a basis, stay tuned, I will test it on the Mainnet, I will post the code

**Comment by oliverw** [comment]
> Today I started the return of Ethereum in my [repository](https://github.com/Konstantin35/miningcore), I took Release 53 as a basis, stay tuned, I will test it on the Mainnet, I will post the code

If you can get it to work, I'll gladly accept a PR to bring it back into `master`.

**Comment by Konstantin35** [comment]
> 
> 
> > Today I started the return of Ethereum in my [repository](https://github.com/Konstantin35/miningcore), I took Release 53 as a basis, stay tuned, I will test it on the Mainnet, I will post the code
> 
> If you can get it to work, I'll gladly accept a PR to bring it back into `master`.

I have created my own repository for my own project and there are minor changes in the code regarding the default name of the worker on all coins. I will create a PR separately if possible and in time.

**Comment by calvintam236** [comment]
@Konstantin35 just a friendly reminder: if you want to make a PR to here, your repo has to be forked from this repo. Otherwise, the PR option is not available. You can branch out at tag `v53` and add your code from there.

**Comment by Konstantin35** [comment]
> 
> 
> @Konstantin35 just a friendly reminder: if you want to make a PR to here, your repo has to be forked from this repo. Otherwise, the PR option is not available. You can branch out at tag `v53` and add your code from there.

Hi I understand I will do this separately.

**Comment by github-vp** [comment]
Reposting my question from gitter..
Using release 50 I build MC on Unix. My miner->MC->Geth are connected and SEEMS to mine. But now I am told ETH is not working since long.
Can someone please confirm if release 50 code base works or not?
Thanks

**Comment by oliverw** [comment]
@github-vp Mining will probably work. It's been a while (years actually) but as far as I remember, the problem starts when blocks are found and need to be accounted for, unlocked and then paid.

**Reply by github-vp** [reply]
Thank you for your response. I am only user on my pool and payment module is disabled. In that case it should be okay for me, right? Assuming “accounting for, unlocking and paying” is part of MC payment module.

**Reply by Konstantin35** [reply]
yes that's right if you disable payouts the blocks will not be updated in the pool they will just remain on the wallet

**Reply by github-vp** [reply]
Thank you. That is sufficient for my need. I have another question, ultimately I am goin to point my rigs (42. 3080s) to the my pool but for now I am doing POC/testing on 2 pc (3080) on each. Both PC are MSI gaming with i9/1 T HDD/32GB ram. One pc runs Geth , MC and another just miner. PC with Geth and MC restarts on its own at least once in 24 hours. Has anyone seen this? As next step I am planning to move Geth to rig and run MC alone and see if problem persists

**Reply by oliverw** [reply]
> Thank you. That is sufficient for my need. I have another question, ultimately I am goin to point my rigs (42. 3080s) to the my pool but for now I am doing POC/testing on 2 pc (3080) on each. Both PC are MSI gaming with i9/1 T HDD/32GB ram. One pc runs Geth , MC and another just miner. PC with Geth and MC restarts on its own at least once in 24 hours. Has anyone seen this? As next step I am planning to move Geth to rig and run MC alone and see if problem persists

I remember geth as an unwieldly beast back when we had an ETH pool at coinfoundry but it didn't crash its docker container at least. Let alone the docker host.

**Reply by github-vp** [reply]
I can’t conclusively say it MC that is causing system crash. Memory dump blame everyone on PC 😁. I will move Geth and see, if not then install MC on another stable PC and see.
BTW I am running MC on WSL running Ubuntu.

**Reply by oliverw** [reply]
> I can’t conclusively say it MC that is causing system crash. Memory dump blame everyone on PC 😁. I will move Geth and see, if not then install MC on another stable PC and see.
> BTW I am running MC on WSL running Ubuntu.

Even though I'm testing under WSL2 from time to time I never used it for production purposes.

**Reply by github-vp** [reply]
Hi guys,
Would you be interested to know how my setup is going on with Geth, MC and miner? If so then let me know and I will share my experience here.

**Reply by oliverw** [reply]
> Hi guys,
> Would you be interested to know how my setup is going on with Geth, MC and miner? If so then let me know and I will share my experience here.

Absolutely!

**Reply by github-vp** [reply]
I have started moving my setup from test PCs to production. PC (HDD)  running Geth and MC use to crash once a day but since moving Geth to production system(SSD) . MC is stable.

**Reply by github-vp** [reply]
Another important observation... on test setup even though GEth download was 50%, Geth allowed mining and matching to block explorer it was allowing mining on latest block. MC reported daemon 100% in sync. On moving Geth to SSD running rig Geth stopped allowing mining and MC reported 50% sync completed

**Reply by github-vp** [reply]
MC does crash...
Geth is switching between syncing and mining allowed mode. Geth size is 794 GB, 50 GB behind today’s size. Mining is allowed on latest block.
MC shows 2 different behaviors:
1: crash with stack trace (something with mapdeamontask ), task could not be completed 
2: MC stuck at “daemon 100% sync “ repeating

**Reply by github-vp** [reply]
```
2021-06-26 18:40:57.1131] [E] [eth] System.AggregateException: One or more errors occurred. (A task was canceled.) ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Miningcore.DaemonInterface.DaemonClient.MapDaemonBatchResponse(Int32 i, Task`1 x) in /mnt/d/mc-unix/miningcore/src
```

**Reply by calvintam236** [reply]
> Another important observation... on test setup even though GEth download was 50%, Geth allowed mining and matching to block explorer it was allowing mining on latest block. MC reported daemon 100% in sync. On moving Geth to SSD running rig Geth stopped allowing mining and MC reported 50% sync completed

Feels like you should open a bug report instead of in this discussion thread about ETC.

**Reply by Konstantin35** [reply]
> 
> 
> MC does crash...
> Geth is switching between syncing and mining allowed mode. Geth size is 794 GB, 50 GB behind today’s size. Mining is allowed on latest block.
> MC shows 2 different behaviors:
> 1: crash with stack trace (something with mapdeamontask ), task could not be completed
> 2: MC stuck at “daemon 100% sync “ repeating

Why are you running the MC on an unsynchronized geth. just run geth so 
`./geth --cache 1024 --port PORT --datadir DATADIR DB ` 
and wait for full synchronization then add 
`--mine --http --http.port RPC_PORT --unlock WALLET --password PASSWORD --allow-insecure-unlock` 
and then run MC

**Reply by github-vp** [reply]
Will do that. Thanks for suggestions. I was running MC on unsync Geth for no particular reason except MC was showing me sync progress

**Reply by github-vp** [reply]
Here is update if you guys are still interested...

1-my Geth node seems to be synced.  Eth.syncing is returning false, ETH.blocknumber is returning current block. One main point is ychart.com indicates size of blockchain is 860 gb but my ethereum folder size is 824 Gb. 
2-I connected MC with Geth using -mine -miner.etherbase -http -http.addr-http.port. I am not using -unlock -password etc as my payment module is disabled and it’s pool requirements not Geth’s. 
3-I have Geth->MC->t-Rex (1 gpu) running and not encountered any issuesso far. When t-Rex is connected to ethermine it show current block being mine, this is not the case when t-Rex is connected to MC
In day or two I will point all my rigs to MC and let you know how it goes. If you have any questions or want me to test any particular thing let me know

**Reply by github-vp** [reply]
Update: 
I have pointed all my rigs to MC. Everything seems working fine. My total hash in MC matches with actual hash generated by my rigs. Blocknumber seems to be the latest. 
One question though... I have provided wallet address to Geth on startup. I have wallet configured in pool as well. Which wallet get the coin first when a block is hit? Assuming it is wallet assigned to Geth than when does coin gets transferred to pool wallet? Does transfer even takes place is payments is disabled.?

**Comment by oliverw** [comment]
Yes all payment related. @Konstantin35 is this correct or did you find anything else?

**Reply by Konstantin35** [reply]
At the moment, ETC does not work because with the latest implementation in ETC [ecip-1099](https://github.com/eth-classic/etchash), you need to add the Etchash module separately

**Reply by oliverw** [reply]
> At the moment, ETC does not work because with the latest implementation in ETC [ecip-1099](https://github.com/eth-classic/etchash), you need to add the Etchash module separately

Add the Etchash to what?

**Reply by Konstantin35** [reply]
Are there any other options?

**Comment by daworm277** [comment]
I would be for the whole project

**Reply by calvintam236** [reply]
Please reply by link. I am not sure who are you replying to.

**Comment by daworm277** [comment]
Ok you gotta explain this to me like imma dummy lol   Sent from Mail for Windows 10 From: github-vpSent: Saturday, June 26, 2021 9:58 AMTo: coinfoundry/miningcoreCc: daworm277; CommentSubject: Re: [coinfoundry/miningcore] ETC Support (#850) Another important observation... on test setup even though GEth download was 50%, Geth allowed mining and matching to block explorer it was allowing mining on latest block. MC reported daemon 100% in sync. On moving Geth to SSD running rig Geth stopped allowing mining and MC reported 50% sync completed—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.

**Reply by calvintam236** [reply]
@daworm277 not sure where is your message in this bunch of text. Please go to the link to reply.

**Comment by konstiantyn** [comment]
I think it is useful to implement ETH/ETC pool.
https://github.com/yinengy/Ethash-C-Implementation
https://github.com/LinzhiChips/libdag

**Reply by n1ckl4s** [reply]
Thanks for the contribution, were you able to do the tests with ETC?

**Comment by konstiantyn** [comment]
How is etc pool going ?

**Comment by MiningCryptoLive** [comment]
any updates on adding etchash??

**Comment by oliverw** [comment]
Have been quite busy on adding other requested functionality with the latest release. Would gladly accept a PR for etchash.

**Reply by konstiantyn** [reply]
> Have been quite busy on adding other requested functionality with the latest release. Would gladly accept a PR for etchash.

Thanks you.

**Comment by n1ckl4s** [comment]
Hi there! 
Could any one add ETC?

**Comment by totom9000** [comment]
We have a working etc and eth code  wil talk with our dev if we shuld post it here.

**Reply by MiningCryptoLive** [reply]
That would be great 👍

**Comment by MataAzi** [comment]
Any updates for ETC?

**Comment by revolutionchain** [comment]
@oliverw I've sent you an email regarding this subject, can you take a look?
